header-rewrite-filter/header_processor.cc

```diff
@@ -32,20 +32,23 @@ namespace HeaderRewriteFilter {
 
         // parse values and call setVals
         try {
-            std::vector<std::string> vals;
-            for(auto it = start + 1; it != operation_expression.end(); ++it) {
-                if (*it == "if") { // condition found
+            std::string val;
+            val = std::string{*(start + 1)};
+            setVal(val);
 
-                    const absl::Status status = HeaderProcessor::ConditionProcessorSetup(operation_expression, it+1); // pass everything after the "if"
+            if ((start + 2) != operation_expression.end()) {
```

Comment:
I see what you're saying, but I was thinking more in terms of readability. If I increment `pos` each time, it's harder to see which exact value I'm accessing from the vector because all the dereferences look like `*pos`. But with `*(pos + 2)` I can take a look at the vector and quickly figure out that the value I was trying to get is two positions away from the beginning.

Let me know if that makes sense or if you feel strongly about incrementing and then accessing!

Reply:
Sure, readability is relative, if you find this easier to work with that's totally fine